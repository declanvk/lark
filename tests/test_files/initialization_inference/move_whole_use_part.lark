//~ execute:no

struct Foo { x: uint }
struct Bar { foo: Foo }

def main() {
  let bar = Bar(foo: Foo(x: 22))
  take_bar(bar)
  take_foo(bar.foo)
  //~ ERROR: access to uninitialized path
}

def take_bar(v: Bar) { }

def take_foo(v: Foo) { }
