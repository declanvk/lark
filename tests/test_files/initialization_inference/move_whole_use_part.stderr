error: access to uninitialized path
- initialization_inference/move_whole_use_part:9:11
9 |   take_foo(bar.foo)
  |            ^^^^^^^
